vignettes/available-theming-variables.Rmd

```diff
@@ -0,0 +1,258 @@
+    # to link to it as it's not available in the table
+    value = str_replace_all(
+      value,
+      "(?<!map-get\\()(\\$[\\w|-]+)",
+      "<span class='dep-link'>\\1</span>"
+    ),
+    section = glue("Section: {section} <a href=\"{section_link}\"><i class=\"fas fa-external-link-alt\"></i></a>"),
+    declaration = glue("<a href=\"{link_to_pos_in_script(line_number)}\"><i class=\"fas fa-external-link-alt\"></i></a>")
+  ) %>%
+  filter(!value_is_function) %>%
+  select(
+    section,
+    variable,
+    value,
+    comment,
+    declaration
+  ) -> variables_df 
+```
+
+
+## Modifying Bootstrap theming variables
+
+While a large amount of customization is possible with the basic parameters in `bs_theme()`, sometimes you may want more fine-grain control. To enable this, `bslib` lets you change any of the [exposed theming variables provided by Bootstrap.](https://getbootstrap.com/docs/4.0/getting-started/theming/) For instance, to adjust the padding amount after each paragraph we can use the variable `paragraph-margin-bottom`...
+
+```{r, eval = FALSE, echo = TRUE}
+# Double space given after each paragraph
+theme <- bs_theme(..., "paragraph-margin-bottom" = "2rem")
+```
+
+For more details on this see the [sass variables section of the theming article](https://rstudio.github.io/bslib/articles/theming.html#sass-variables-1) and the [original Bootstrap documentation of theme variables.](https://getbootstrap.com/docs/4.0/getting-started/theming/)
+
+
+
+## All theming variables
+
+Below is a table that shows all the available themeing variables along with basic comments on that variable's functionality as provided in the original Bootstrap code. 
+_Note: these variables are specifically for Bootstrap 4 and many won't work for Bootstrap 3._
+
+### How to use table
+
+- Each section header corresponds to a section within the source file denoted by comments. Click on the link icon in the header to be taken to the start of that section in the variable declaration script. 
+- Click on dependency variable within the value column to be taken to the declaration of that dependency in the table.
+- The __Find in source__ column will take you to the position in the original `.scss` file where the variable was declared. The context may help clarify the variables role. 
+
+
+```{r}
+# This chunk of JS requires the DataTable object to be available in scope as "table" 
+js_helper_fns <- "
+function find_and_select_variable(table, desired_var, update_page) {
+  var row_index = table
+    .column(1, { order: 'current' })
+    .data()
+    .indexOf(desired_var);
+  if (row_index !== -1) {
+    table.rows({ selected: true }).deselect();
+    table.row(row_index).select();
+    location.hash = desired_var.replace('$', '#');
+
+    if (update_page) {
+      table.row(row_index).scrollTo();
+    }
+  }
+}
+"
+
+variables_df %>%
+  rename(
+    Variable = variable,
+    Value = value,
```

Comment:
I think we may want to consider having both a `Sass value` (i.e., the current `Value` column) and `CSS value` (which would be `bslib::bs_get_variables(bslib::bs_theme(), variable)`)

Reply:
Somewhat relatedly, it'll a little unfortunate that you currently have to scroll far to the right to get the source links, perhaps it'd be better to column them in the `Variable` column (or at least closer to it?)

Reply:
![image](https://user-images.githubusercontent.com/6764693/106166679-6edf2200-615a-11eb-8d1b-5edc3736ac78.png)

Now the compiled css value can be seen on hover of the sass value using a tooltip. I think it's kinda slick.